src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -0,0 +1,177 @@
+""" The MSSQL T-SQL dialect.
+
+https://docs.microsoft.com/en-us/sql/t-sql/language-elements/language-elements-transact-sql
+
+
+"""
+from enum import Enum
+from typing import Generator, List, Tuple, NamedTuple, Optional, Union
+
+from sqlfluff.core.parser import (
+    Matchable,
+    BaseSegment,
+    KeywordSegment,
+    SymbolSegment,
+    Sequence,
+    GreedyUntil,
+    StartsWith,
+    OneOf,
+    Delimited,
+    Bracketed,
+    AnyNumberOf,
+    Ref,
+    SegmentGenerator,
+    Anything,
+    Indent,
+    Dedent,
+    Nothing,
+    OptionallyBracketed,
+    StringLexer,
+    RegexLexer,
+    CodeSegment,
+    CommentSegment,
+    WhitespaceSegment,
+    NewlineSegment,
+    StringParser,
+    NamedParser,
+    RegexParser,
+    Conditional,
+)
+
+from sqlfluff.core.dialects.base import Dialect
+from sqlfluff.core.dialects.common import AliasInfo
+from sqlfluff.core.parser.segments.base import BracketedSegment
+
+from sqlfluff.dialects.ansi_keywords import (
+    ansi_reserved_keywords,
+    ansi_unreserved_keywords,
+)
+from sqlfluff.dialects.tsql_keywords import (
+    # BARE_FUNCTIONS,
+    RESERVED_KEYWORDS,
+    # UNRESERVED_KEYWORDS,
+)
```

Comment:
Lots of these aren't used - which is upsetting the black test that you're bothering to import it.

```suggestion
from sqlfluff.core.parser import (
    BaseSegment,
    Sequence,
    GreedyUntil,
    OneOf,
    Bracketed,
    Ref,
    Anything,
    RegexLexer,
    CodeSegment,
    RegexParser,
)
```

(Note: I realise some of this is work in progress but I think better to get clean runs and expand, so you don't lose track of other issues).

Reply:
I use visual code so can easily remove the not-used classes when I'm finished.